### PR TITLE
Better semantic version for main builds

### PR DESCRIPTION
**Pull request by himadrisingh**
Please go the the `Preview` tab and select the appropriate sub-template:

* [Applications](?expand=1&template=applications_template.md)
* [Others](?expand=1&template=fallback_template.md)

### CODE REVIEW

**Review by nishantmonu51 — APPROVED** [review]
LGTM

**Review by Kavinjsir — APPROVED** [review]
Looks good to me.

Just one question: What is the use case for us to add the SHA as a suffix after a semver?

File: .github/workflows/rill-cloud.yml [hunk 1]
```diff
@@ -43,16 +43,19 @@ jobs:
...
           gcloud auth configure-docker
 
-          docker build -t gcr.io/rilldata/rill-headless:${GITHUB_SHA} .
-          docker push gcr.io/rilldata/rill-headless:${GITHUB_SHA}
+          docker build -t gcr.io/rilldata/rill-headless:${VERSION} .
+          docker push gcr.io/rilldata/rill-headless:${VERSION}
 
           if [ ${RELEASE} == "true" ]; then
-            docker tag gcr.io/rilldata/rill-headless:${GITHUB_SHA} gcr.io/rilldata/rill-headless:${GITHUB_REF_NAME}
+            docker tag gcr.io/rilldata/rill-headless:${VERSION} gcr.io/rilldata/rill-headless:${GITHUB_REF_NAME}
```
Comment (kaspersjo) [inline .github/workflows/rill-cloud.yml:58]: Would be great if we could add a release tag without the build metadata for the official release images in addition to the one with build metadata.

**Review by kaspersjo — CHANGES_REQUESTED** [review]

File: .github/workflows/rill-cloud.yml (hunk 1 above)
Comment (himadrisingh) [reply .github/workflows/rill-cloud.yml:58]: this is for release tag only, GITHUB_REF_NAME would be tag that we create.

**Review by kaspersjo — APPROVED** [review]

**Pull request merged**